pulsar-broker/src/test/java/org/apache/pulsar/client/impl/TransactionEndToEndTest.java

```diff
@@ -757,12 +758,31 @@ public void produceAndConsumeCloseStateTxnTest() throws Exception {
             timeoutTxn.commit().get();
             fail();
         } catch (Exception e) {
-            assertTrue(e.getCause() instanceof TransactionNotFoundException);
+            assertTrue(e.getCause() instanceof TransactionCoordinatorClientException.InvalidTxnStatusException);
         }
         Field field = TransactionImpl.class.getDeclaredField("state");
         field.setAccessible(true);
         TransactionImpl.State state = (TransactionImpl.State) field.get(timeoutTxn);
-        assertEquals(state, TransactionImpl.State.ERROR);
+        assertEquals(state, TransactionImpl.State.TIMEOUT);
```

Comment:
please don't change this test, we should check the TC timeout, so return State.ERROR